### ISSUE TITLE
Form fields have ARIA invalid entry

### ISSUE THREAD

**Issue by pollecuttn** [wellcomecollection/wellcomecollection.org#10229]
**What**
Various form fields across the site said ARIA invalid when you first get to them

**Where**

Form fields e.g. login page, account sign up, newsletter sign up

**TODO**

- [ ] Improve ARIA properties on form fields

**Background**

From the Hassell Inclusion accessibility audit August 2023.

Dovetail: https://proficient-wallaby-dpwd.dovetailapp.com/insights/Form-fields-have-ARIA-invalid-entry-2lxxERcdj1O6PrAmi0yhvL
WCAG fail

Miro board: https://miro.com/app/board/uXjVMjsL5YE=/

**Comment by davidpmccormick** [comment]
During the audit, we were shown the screenreader NVDA announcing 'invalid entry' on text inputs that had not yet been interacted with.

This announcement was present for inputs that we have created ourselves (`TextInput`) as well as those from a third-party, Auth0, over which we have little-to-no control.

We deliberately add `aria-invalid` to our `TextInput` component, but only if the input has lost focus after there has at least been _some_ interaction. So we would expect to hear 'invalid entry' under those conditions. Checking in dev-tools, the Auth0 inputs don't appear to get an `aria-invalid` attribute under any circumstances, so I'm unsure what could have been the cause of the announcement in the audit for those.

I've tested both of these types of input using NVDA and haven't been able to reproduce either issue – the screen recordings below show what we would expect to happen.

__NewsletterPromo__ (uses `TextInput`)
https://github.com/wellcomecollection/wellcomecollection.org/assets/1394592/683ef182-8e26-4320-b970-1c44a9d559af

__Auth0__
https://github.com/wellcomecollection/wellcomecollection.org/assets/1394592/0bcf66e0-ea0b-419a-a9c9-cafa87fea269

The consistent error across first- and third-party inputs maybe suggests that the issue might have been due to the auditor's set-up. I think we can maybe close as can't reproduce unless we can get any more info.

**Comment by davidpmccormick** [comment]
I narrowed it down to the `required` attribute (hence why it was also a problem for the Auth0 inputs). When it's present, NVDA will announce 'invalid entry' in Firefox, regardless of whether `aria-invalid` exists, or is set to `false`. [This SO post](https://stackoverflow.com/questions/20386266/how-to-prevent-jaws-from-saying-invalid-entry-on-required-fields) corroborates what we've found. There's a mention of using `aria-required` instead/as well so I tested with those.

- Using `required` and `aria-required` together doesn't make any difference (still announces 'invalid entry')
- Using `aria-required` alone cleans up what the screenreader announces, but doesn't prevent form submission when the field is empty (which is one of the key reasons for using `required` in the first place)

I think having the 'invalid entry' announced in Firefox with NVDA is the lesser of two evils compared with losing the prevent-invalid-submission behaviour. Seems like a browser/screenreader issue that we should probably not try to work around.

I don't want to close this before I get other opinions though.
